sotopia/generation_utils/generate.py

```diff
@@ -334,7 +334,27 @@ def obtain_chain(
             openai_api_base="https://api.groq.com/openai/v1",
             openai_api_key=os.environ.get("GROQ_API_KEY"),
         )
-        chain = LLMChain(llm=chat_openai, prompt=chat_prompt_template)
+        chain = chat_prompt_template | chat_openai
+        return chain
+    elif "azure" in model_name:
+        # azure|deployment_name|version|endpoint|azure_openai_api_key
```

Comment:
maybe the api_key can just be an environment variable?

Reply:
but what if the user wants to use different azure models with different api keys, then this wouldn't work

Reply:
Yeah, this is tricky. Ideally, we should have a `LLMConfig` pydantic baseclass where the API key is a pydantic `SecretStr`.

Reply:
Can you use the deployments with the same apikey first? I can implement the above feature with a new PR.